server/order.js

```diff
@@ -0,0 +1,138 @@
+/* Orders is observer of: Inventory
+ *           observed by: Drivers
+ */
+class Orders {
+    constructor() {
+        this.orders = [];
+        this.drivers = [];
+    }
+
+    fillList(ordersList) {
+        ordersList.forEach(order => {
+            order = new Order(order);
+            if (!this.orders.includes(order)) {
+                this.orders.push(order);
+            }
+        });
+    }
+
+    // Beginning of Observable's duties
+    notify(URL) {
+        try {
+            let fbServer = new XMLHttpRequest();
+            fbServer.open("POST", foodBankURL, true);
+            fbServer.setRequestHeader('Content-Type', 'application/json');
+            fbServer.send(JSON.stringify({
+                "orderId": 11111,
+                "foodBankId": 3,
+                "groceryId": 1,
+                "Inventory": [{
+                        "inventoryItemId": 333,
+                        "name": "Orange",
+                        "quantity": 20,
+                        "expiryDate": new Date("2020-03-28"),
+                        "groceryStoreId": 1
+                    },
+                    {
+                        "inventoryItemId": 334,
+                        "name": "Apple",
+                        "quantity": 20,
+                        "expiryDate": new Date("2020- 03 - 20"),
+                        "groceryStoreId": 1
+                    }
+                ]
+            }))
+
+            fbServer.onreadystatechange = function () {
+                if (this.status === 200) {
+                    console.log("200");
+                }
+            };
+
+            fbServer.send();
+        } catch (error) {
+            console.log("Can't notify entity.")
+        }
+    }
+
+    notifyDriver(context) { // push notification
+        try {
+            console.log('Driver notified with ' + context);
+        } catch (error) {
+            console.log("Notification failed" + error);
+        }
+    }
+
+    addDriver(driver) { // adds observer
+        try {
+            if (this.isDriverValid(driver)) {
+                this.diver.push(driver);
+                console.log("Driver added.");
+            }
+        } catch (error) {
+            console.log("Can't assign driver" + error);
+        }
+
+    }
+
+    removeDriver(driverId) { // remove observer
+        try {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary block.](https://app.codacy.com/gh/CSC492-DSC-Community-Projects/Backend/pullRequest?prid=4855140)

Reply:
hmm this is needed for unit tests. Should I comment it out for now?